src/components/Account/ScreenConnected.js

```diff
@@ -33,15 +30,15 @@ function AccountScreenConnected({ providerId, onClosePopover }) {
   const providerInfo = getProviderFromUseWalletId(providerId)
 
   const goToProfile = useCallback(() => {
-    history.push(`/profile`)
+    router.push(`/profile`)
     onClosePopover()
-  }, [history, onClosePopover])
+  }, [router, onClosePopover])
 
   const goToStakeManagement = useCallback(() => {
-    const path = buildGardenPath(history.location, 'collateral')
-    history.push(path)
+    const path = buildGardenPath(router, 'collateral')
```

Comment:
When we build the path using `buildGardenPath` we don't replace the correct parameters. The final URL looks something like: `[networkType]/garden/[gardenAddress]/proposal/1`.

Instead, the `GardenList` component as we are doing it manually is working fine: https://github.com/reznord/gardens-ui/blob/staging/src/components/GardensList.js#L80-L84

Reply:
Yes, I am removing the whole `buildGardenPath` dependency entirely.

I have migrated almost all the places where we are using the `buildGardenPath` but there are a few places which I am stuck at (will ping you on discord for this).